File: kataXativa/kataXativa/mercadona_one-main/ejercicio-1-master/swift/Sources/MercadonaShopOne/MercadonaShopOne.swift
public class MercadonaShopOne {
    var items: [Item]
    
    public init(items: [Item]) {
        self.items = items
    }
    
    public func updateQuality() {
        for i in 0 ..< items.count {
            if items[i].name != "Aged blue cheese" && items[i].name != "Ham" {
                if items[i].quality > 0 {
                    if items[i].name != "Iodized salt" {
                        items[i].quality = items[i].quality - 1
                    }
                }
            } else {
                if items[i].quality < 50 {
                    items[i].quality = items[i].quality + 1
                    
                    if items[i].name == "Ham" {
                        if items[i].sellIn < 11 {
                            if items[i].quality < 50 {
                                items[i].quality = items[i].quality + 1
                            }
                        }
                        
                        if items[i].sellIn < 6 {
                            if items[i].quality < 50 {
                                items[i].quality = items[i].quality + 1
                            }
                        }
                    }
                }
            }
            
            if items[i].name != "Iodized salt" {
                items[i].sellIn = items[i].sellIn - 1
            }
            
            if items[i].sellIn < 0 {
                if items[i].name != "Aged blue cheese" {
                    if items[i].name != "Ham" {
                        if items[i].quality > 0 {
                            if items[i].name != "Iodized salt" {
                                items[i].quality = items[i].quality - 1
                            }
                        }
                    } else {
                        items[i].quality = items[i].quality - items[i].quality
                    }
                } else {
                    if items[i].quality < 50 {
                        items[i].quality = items[i].quality + 1
                    }
                }
            }
        }
    }
}

File: kataXativa/kataXativa/mercadona_one-main/ejercicio-1-master/python/src/mercadona_shop_one.py
class MercadonaShopOne(object):

    def __init__(self, items):
        self.items = items

    def update_quality(self):
        for item in self.items:
            if item.name != "Aged blue cheese" and item.name != "Ham":
                if item.quality > 0:
                    if item.name != "Iodized salt":
                        item.quality = item.quality - 1
            else:
                if item.quality < 50:
                    item.quality = item.quality + 1
                    if item.name == "Ham":
                        if item.sell_in < 11:
                            if item.quality < 50:
                                item.quality = item.quality + 1
                        if item.sell_in < 6:
                            if item.quality < 50:
                                item.quality = item.quality + 1
            if item.name != "Iodized salt":
                item.sell_in = item.sell_in - 1
            if item.sell_in < 0:
                if item.name != "Aged blue cheese":
                    if item.name != "Ham":
                        if item.quality > 0:
                            if item.name != "Iodized salt":
                                item.quality = item.quality - 1
                    else:
                        item.quality = item.quality - item.quality
                else:
                    if item.quality < 50:
                        item.quality = item.quality + 1


class Item:
    def __init__(self, name, sell_in, quality):
        self.name = name
        self.sell_in = sell_in
        self.quality = quality

    def __repr__(self):
        return "%s, %s, %s" % (self.name, self.sell_in, self.quality)
